cogs/highlight.py

```diff
@@ -0,0 +1,209 @@
+                await self._sleep_then_delete(t_msg,5)
+        else:
+            new_user = {}
+            new_user['id'] = ctx.message.author.id
+            new_user['words'] = [word]
+            self.highlights['guilds'][guild_idx][guild_id]['users'].append(new_user)
+            self._update_highlights(self.highlights)
+        
+    @highlight.command(name="remove", pass_context=True, no_pm=True)
+    async def remove_highlight(self, ctx, word: str):
+        guild_id = ctx.message.server.id
+        user_id = ctx.message.author.id
+        user_name = ctx.message.author.name
+        
+        guild_idx = self._check_guilds(guild_id)
+        user = self._is_registered(guild_idx,guild_id,user_id)
+        
+        if user is not None:
+            user_idx = user[0]
+            user_rm = user[1]
+            if word in user_rm['words']:
+                user_rm['words'].remove(word)
+                self.highlights['guilds'][guild_idx][guild_id]['users'][user_idx] = user_rm
+                self._update_highlights(self.highlights)
+                t_msg = await self.bot.say("Highlight word removed, {}".format(user_name))
+                await self._sleep_then_delete(t_msg,2)
+            else:
+                t_msg = await self.bot.say("Sorry {}, you don't have this word highlighted".format(user_name))
+                await self._sleep_then_delete(t_msg,5)
+        else:
+            msg = "Sorry {}, you aren't currently registered for highlights."
+            msg += " Add a word to become registered"
+            t_msg = await self.bot.say(msg.format(user_name))
+            await self._sleep_then_delete(t_msg,5)
+        
+    @highlight.command(name="list", pass_context=True, no_pm=True)
+    async def list_highlight(self, ctx):
+        guild_id = ctx.message.server.id
+        user_id = ctx.message.author.id
+        user_name = ctx.message.author.name
+        
+        guild_idx = self._check_guilds(guild_id)
+        user = self._is_registered(guild_idx,guild_id,user_id)
+        
+        if user is not None:
+            user_list = user[1]
+            if len(user_list['words']) > 0:
+                msg = ""
+                for word in user_list['words']:
+                    msg += word
+                    msg += "\n"
+                    
+                embed = discord.Embed(description=msg,colour=discord.Colour.red())
+                embed.set_author(name=ctx.message.author.name,icon_url=ctx.message.author.avatar_url)
+                t_msg = await self.bot.say(embed=embed)
+                await self._sleep_then_delete(t_msg,5)
+            else:
+                t_msg = await self.bot.say("Sorry {}, you have no highlighted words currently".format(user_name))
+                await self._sleep_then_delete(t_msg,5)
+            
+    @highlight.command(name="import", pass_context=True, no_pm=False)
+    async def import_highlight(self, ctx, from_server: str):

```

Comment:
yuppo! this will just be an ease-of-use util, where someone can import words if they join a server that has the bot or something. gonna push this to lui-cogs eventually so i am thinking outside the scope of just ren-bot for SAD